improver/cli/__init__.py

```diff
@@ -239,27 +242,29 @@ def constrained_inputcubelist_converter(to_convert):
                 than one match is found.
         """
         from improver.utilities.load import load_cube
+        from iris import load_raw
         from iris.cube import CubeList
-
-        cubelist = CubeList()
-        for constr in constraints:
-            constr_list = [constr] if isinstance(constr, str) else constr
-            found_cubes = []
-            for constr_item in constr_list:
-                try:
-                    found_cubes.append(maybe_coerce_with(
-                        load_cube, to_convert, constraints=constr_item))
-                except ValueError:
-                    pass
-            if len(found_cubes) != 1:
-                msg = (f"Incorrect number of valid inputs available for the "
-                       "{constr} constraint. "
-                       f"Number of valid inputs: {len(found_cubes)} "
-                       f"The valid inputs found are: {found_cubes}")
-                raise ValueError(msg)
-            cubelist.extend(found_cubes)
-
-        return cubelist
+        # Iris will add on an extra cube which isn't to be counted.
+        meta_len = 1
+        cube_length = len(maybe_coerce_with(load_raw, to_convert)) - meta_len
+        partial_match = False
+        for constraints in constraint_lists:
+            try:
+                cubes = CubeList(
+                    maybe_coerce_with(load_cube, to_convert, constraints=j)
+                    for j in constraints)
+                # If there is a match but there are more cubes in the cubelist.
+                if len(cubes) == cube_length:
+                    return cubes
+                partial_match = True
+            except ValueError:
+                # Unable to load a cube with all constraints. Try next list.
+                pass
+        if partial_match:
+            raise ValueError("Partial match found, cubes must be a whole list."
+                             f"Cubes must be called: {constraint_lists}")
+        raise ValueError("Not all cubes matching the required names."
+                         f"Cubes must be called: {constraint_lists}")
```

Comment:
I think we need to revisit these error messages.  And what is the condition for raising the second error?